### PR TITLE
feat(dropdown): react component

### PR DESCRIPTION
**Pull request by arnaudvalle**
`overToggle` option needs implementing (need to discuss this with Guillaume to see if Popover is going to handle that in the end or not).

Also not sure it handles all the cases of the AngularJS version yet (in terms of positioning etc) but will have to wait for Matthias to test / confirm this.

### CODE REVIEW

File: src/components/dropdown/react/Dropdown.tsx [hunk 1]
```diff
@@ -64,12 +85,63 @@ const DEFAULT_PROPS: IDefaultPropsType = {};
...
-    ...props
-}: DropdownProps): React.ReactElement => (
-    <div className={classNames(className, handleBasicClasses({ prefix: CLASSNAME }))} {...props}>
-        {children}
-    </div>
-);
+    closeOnClick = DEFAULT_PROPS.closeOnClick,
+    escapeClose = DEFAULT_PROPS.escapeClose,
+    offset,
+    // overToggle = DEFAULT_PROPS.overToggle,
```
Comment (Flo5k5) [inline src/components/dropdown/react/Dropdown.tsx]: Remove if useless

File: src/components/dropdown/react/Dropdown.tsx [hunk 2]
```diff
@@ -64,12 +85,63 @@ const DEFAULT_PROPS: IDefaultPropsType = {};
...
+    position,
+    toggleElement,
+    width,
+}: DropdownProps): React.ReactElement => {
+    const wrapperRef: React.RefObject<HTMLDivElement> = useRef(null);
+    const [isOpen, setIsOpen]: [boolean, (isOpen: boolean) => void] = useState(Boolean(false));
```
Comment (Flo5k5) [inline src/components/dropdown/react/Dropdown.tsx]: I think you can use `useState<boolean>(false)` instead

File: src/components/dropdown/react/Dropdown.tsx [hunk 3]
```diff
@@ -64,12 +85,63 @@ const DEFAULT_PROPS: IDefaultPropsType = {};
...
+        if (!closeOnClick) {
+            return;
+        }
+        closeDropdown();
+    });
+
+    return (
+        <div
+            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME }), {
+                [`${CLASSNAME}--has-toggle`]: toggleElement,
```
Comment (Flo5k5) [inline src/components/dropdown/react/Dropdown.tsx]: You could use `hasToggle: toggleElement` instead, just after prefix.

File: demo/react/components/dropdown/default.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            <ListItem size={ListItemSizes.tiny}>Trondheim</ListItem>
+        </List>
+    );
+};
+
+const createComplexMenuList: (setIsOpen: (isOpen: boolean) => void) => ReactNode = (
+    setIsOpen: (isOpen: boolean) => void,
+): ReactNode => {
+    // tslint:disable-next-line no-unused
+    const onItemSelectedHandler: (item: ListItem) => void = (item: ListItem): void => {
```
[the remark this answers is not in the document]
Comment (arnaudvalle) [reply demo/react/components/dropdown/default.tsx]: It was to show in the demo that we were getting the selected item. I'll add a `console.log` so it's a bit more clear.

File: src/components/dropdown/react/Dropdown.tsx (hunk 1 above)
Comment (arnaudvalle) [reply src/components/dropdown/react/Dropdown.tsx]: I'll remove for now but as discussed in the comment, Guillaume is going to add the option to the popover and I'll add it here as well once it's done.

File: src/components/dropdown/react/Dropdown.tsx (hunk 2 above)
Comment (arnaudvalle) [reply src/components/dropdown/react/Dropdown.tsx]: ok - I just saw it the way I did it somewhere else in the project so replicated that. I'm fine with either tbh.

File: src/components/dropdown/react/Dropdown.tsx (hunk 3 above)
Comment (arnaudvalle) [reply src/components/dropdown/react/Dropdown.tsx]: 👍 yup I need to refactor this now that you've tweaked the helper

**Comment by arnaudvalle** [comment]
@Flo5k5 should be ok now (and test should pass hopefully)

**Comment by matmkian** [comment]
Something went wrong with this demo
ReferenceError: useClickAway is not defined

Can't test :-(

**Comment by matmkian** [comment]
Cannot test: 
Something went wrong with this demo
TypeError: Cannot read property 'BOTTOM_START' of undefined

**Comment by Flo5k5** [comment]
> Cannot test:
> Something went wrong with this demo
> TypeError: Cannot read property 'BOTTOM_START' of undefined

I haven't got any error when launching DS, are you using the last version of this branch?

File: demo/style/layout/main/_main.scss [hunk 5]
```diff
@@ -2,6 +2,12 @@
    Main
    ========================================================================== */
 
+input:focus,
```
Comment (matmkian) [inline demo/style/layout/main/_main.scss]: Possible d'enlever ça, ça pique les eux :-)

Comment (Flo5k5) [reply demo/style/layout/main/_main.scss]: 👍

**Comment by matmkian** [comment]
Test ok !

**Pull request merged**